app/models/exporters/usage_report_exporter.rb

```diff
@@ -0,0 +1,169 @@
+# frozen_string_literal: true
+
+# Export forms to an Excel file (.xlsx)
+# rubocop:disable Metrics/ClassLength
+class Exporters::UsageReportExporter < Exporters::BaseExporter
+  attr_accessor :file_name, :workbook, :errors, :worksheets
+
+  class << self
+    def id
+      'xlsx'
+    end
+
+    def supported_models
+      [User]
+    end
+  end
+
+  def initialize(output_file_path = nil, config = {}, options = {})
+    super(output_file_path, config, options)
+    self.workbook = WriteXLSX.new(buffer)
+    self.worksheets = {}
+    self.locale = user&.locale || I18n.locale
+  end
+
+  def export(start_date, end_date, request)
+    export_user_row(start_date, end_date, request)
+    modules = UsageReport.modules
+    modules.each { |modul| export_modules(modul, start_date, end_date) }
+  end
+
+  def complete
+    @workbook.close
+    buffer
+  end
+
+  private
+
+  def export_modules(modul, start_date, end_date)
+    return if modul.blank? || modul.instance_of?(ActiveRecord::Relation)
+
+    export_module_to_workbook(modul, start_date, end_date)
+  end
+
+  def export_module_to_workbook(modul, start_date, end_date)
+    worksheet = workbook.add_worksheet(modul.name)
+    adjust_column_width(worksheet)
+    worksheet.write(0, 0, module_header(modul.name))
+    worksheet.write(1, 0, module_content(modul.unique_id, start_date, end_date, modul.name))
+  end
+
+  def module_header(modul_name)
+    common_keys = [modul_name, 'Total Cases', 'Open Cases', 'Closed Cases', 'Open this quarter',
+                   'Closed this Quarter', 'Total Services', 'Total incidents', 'Incidents this quarter']
+    case modul_name
+    when 'MRM'
+      [modul_name, 'Total incidents', 'Incidents this quarter']
+    when 'GBV'
+      common_keys
+    else
+      common_keys + ['Total followups']
+    end
+  end
+
+  def get_common_keys(module_id, start_date, end_date)
+    ['', UsageReport.total_records(module_id, Child).count, UsageReport.open_cases(module_id).count,
```

Comment:
This is the wrong place to do the `.count` query. It needs to take place on the UsageReport